specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -73,36 +73,41 @@ export const nonUniqueBusinessRuleDefs: MappedBusinessRuleDefs = {
         const returned = borrowMaterial.get('quantityReturned');
         const resolved = borrowMaterial.get('quantityResolved');
         const quantity = borrowMaterial.get('quantity');
-        var newVal: number | undefined = undefined;
-        if (
+
+        const adjustedReturned =
           typeof quantity === 'number' &&
           typeof returned === 'number' &&
-          returned > quantity
-        ) {
-          newVal = quantity;
-        }
-        if (returned && resolved && returned > resolved) {
-          newVal = resolved;
-        }
+          typeof resolved === 'number'
+            ? returned > quantity
+              ? quantity
+              : returned > resolved
+              ? resolved
+              : returned
+            : undefined;
 
-        newVal && borrowMaterial.set('quantityReturned', newVal);
+        if (typeof adjustedReturned === 'number')
+          borrowMaterial.set('quantityReturned', adjustedReturned);
       },
       quantityResolved: (
         borrowMaterial: SpecifyResource<BorrowMaterial>
       ): void => {
         const resolved = borrowMaterial.get('quantityResolved');
         const quantity = borrowMaterial.get('quantity');
         const returned = borrowMaterial.get('quantityReturned');
-        let newVal: number | undefined = undefined;
-        if (resolved && quantity && resolved > quantity) {
-          newVal = quantity;
-        }
-        if (resolved && returned && resolved < returned) {
-          newVal = returned;
-        }
 
-        if (typeof newVal === 'number')
-          borrowMaterial.set('quantityResolved', newVal);
+        const adjustedResolved =
```

Comment:
(optional) this function and the one before it are very similar. if it's not too complicated, see if parts of it or entire thing can be abstracted away into a function